File: ejercicio4/file_reader.rb
# frozen_string_literal: true

class FileReader
  def self.read_cases(file_path)
    cases = Hash.new { |hash, key| hash[key] = [] }
    current_case = nil
    File
      .open(file_path)
      .each do |line|
        if line.start_with?('#')
          current_case = line[/\d+/].to_i
        elsif !line.strip.empty?
          cases[current_case] << line.strip # Add the current case to the hash for the current_case number
        end
      end
    cases.to_a
  end
end

File: ejercicio4/file_reader.py
class FileReader:
    @staticmethod
    def read_cases(file_path):
        cases = {}
        current_case = None
        with open(file_path) as file:
            for line in file:
                if line.startswith('#'):
                    # grab last word in line
                    current_case = line.strip().split()[-1]
                elif line.strip():
                    cases.setdefault(current_case, []).append(line.strip())
        return list(cases.items())
